File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        vector<int> empty;
        int ans =0;
         generate(empty,0,nums,ans);
        return ans;
    }
private:
    void generate(vector<int> &empty,int i,vector<int> &nums,int &ans){
        int sum = 0;
        for(auto x: empty)
            sum = sum^x;
       ans = ans+sum;
        
        for(i;i<nums.size();i++){
            empty.push_back(nums[i]);
            
            generate(empty,i+1,nums,ans);
            empty.pop_back();
        }
        
    }
};

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
        int sum = 0;
        for(int i=0;i<(1<<nums.length);i++){
            int temp = 0;
            for(int j=0;j<nums.length;j++){
                if((i & (1<<j))>0){
                    temp ^= nums[j];
                }
            }
            sum += temp;
        }
        return sum;
    }
}